File: tests/src/org.apache.pekko/pekko-protobuf-v3_3/1.0.1/src/test/scala/org_apache_pekko/pekko_protobuf_v3_3/InternalTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_pekko.pekko_protobuf_v3_3

import org.apache.pekko.protobufv3.internal.Internal
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class InternalTest {
  @Test
  def getDefaultInstanceUsesMessageStaticAccessor(): Unit = {
    val defaultInstance: GeneratedMessageLiteSerializedFormProbe =
      Internal.getDefaultInstance(classOf[GeneratedMessageLiteSerializedFormProbe])

    assertThat(defaultInstance).isSameAs(GeneratedMessageLiteSerializedFormProbe.getDefaultInstance())
  }
}

File: tests/src/org.apache.hadoop.thirdparty/hadoop-shaded-protobuf_3_7/1.1.1/src/test/java/org_apache_hadoop_thirdparty/hadoop_shaded_protobuf_3_7/InternalTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_hadoop_thirdparty.hadoop_shaded_protobuf_3_7;

import static org.assertj.core.api.Assertions.assertThat;

import org.apache.hadoop.thirdparty.protobuf.Internal;
import org.apache.hadoop.thirdparty.protobuf.MessageLite;
import org.apache.hadoop.thirdparty.protobuf.StringValue;
import org.junit.jupiter.api.Test;

public class InternalTest {
    @Test
    void getDefaultInstanceReturnsRegisteredMessageDefault() {
        MessageLite defaultInstance = Internal.getDefaultInstance(StringValue.class);

        assertThat(defaultInstance).isSameAs(StringValue.getDefaultInstance());
    }
}
